tests/epics/adcore/test_cont_acq_detector.py

```diff
@@ -0,0 +1,175 @@
+import pytest
+
+from ophyd_async.core import DetectorTrigger, PathProvider, TriggerInfo, init_devices
+from ophyd_async.epics import adcore
+from ophyd_async.testing import set_mock_value
+
+
+@pytest.fixture
+def cont_acq_det(ad_standard_det_factory):
+    det = ad_standard_det_factory(adcore.ContAcqAreaDetector)
+
+    # Set a few additional settings specific to cont acq detectors
+    set_mock_value(det.driver.acquire, True)
+    set_mock_value(det.cb_plugin.array_size_x, 10)
+    set_mock_value(det.cb_plugin.array_size_y, 10)
+
+    return det
+
+
+@pytest.fixture
+def cont_acq_controller(RE):
+    with init_devices(mock=True):
+        drv = adcore.ADBaseIO("DRV")
+        cb_plugin = adcore.NDPluginCBIO("CB1")
+
+    set_mock_value(drv.image_mode, adcore.ADImageMode.CONTINUOUS)
+    set_mock_value(drv.acquire_time, 0.8)
+    set_mock_value(drv.acquire_period, 1.0)
+    set_mock_value(drv.acquire, True)
+    return adcore.ADBaseContAcqController(drv, cb_plugin)
+
+
+@pytest.fixture
+def one_shot_trigger_info_factory():
+    def generate_one_shot_trigger_info(
+        trigger_mode=DetectorTrigger.INTERNAL, livetime=0.8
+    ):
+        if trigger_mode != DetectorTrigger.INTERNAL:
+            return TriggerInfo(
+                number_of_triggers=1,
+                trigger=trigger_mode,
+                livetime=livetime,
+                deadtime=0.001,
+            )
+        else:
+            return TriggerInfo(
+                number_of_triggers=1, trigger=trigger_mode, livetime=livetime
+            )
+
+    return generate_one_shot_trigger_info
```

Comment:
I'm not a fan of wrapping functions as fixtures, not sure it adds anything...
```suggestion
def generate_one_shot_trigger_info(
    trigger_mode=DetectorTrigger.INTERNAL, livetime=0.8
):
        if trigger_mode != DetectorTrigger.INTERNAL:
            return TriggerInfo(
                number_of_triggers=1,
                trigger=trigger_mode,
                livetime=livetime,
                deadtime=0.001,
            )
        else:
            return TriggerInfo(
                number_of_triggers=1, trigger=trigger_mode, livetime=livetime
            )
```